tests/h/services/search_index/_queue_test.py

```diff
@@ -137,206 +136,194 @@ def add_where(self, queue):
+        counts = queue.sync(1)
 
         assert counts == {}
         batch_indexer.index.assert_not_called()
 
     def test_it_ignores_jobs_that_arent_scheduled_yet(
-        self, add_all, batch_indexer, queue
+        self, batch_indexer, factories, now, queue
     ):
-        add_all(schedule_in=ONE_WEEK_IN_SECONDS)
+        factories.SyncAnnotationJob(scheduled_at=now + datetime_.timedelta(hours=1))
 
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {}
         batch_indexer.index.assert_not_called()
 
-    @pytest.mark.usefixtures("with_queued_annotations")
-    def test_it_ignores_jobs_beyond_limit(
-        self, all_annotation_ids, batch_indexer, queue
-    ):
-        queue.sync(LIMIT)
+    def test_it_ignores_jobs_beyond_limit(self, batch_indexer, factories, queue):
+        limit = 1
+        factories.SyncAnnotationJob.create_batch(size=limit + 1)
+
+        queue.sync(limit)
 
-        for annotation_id in all_annotation_ids[LIMIT:]:
-            assert annotation_id not in batch_indexer.index.call_args[0][0]
+        assert len(batch_indexer.index.call_args[0][0]) == limit
 
     def test_it_ignores_jobs_that_are_expired(
         self, batch_indexer, db_session, factories, now, queue
     ):
-        annotation = factories.Annotation()
-        queue.add_by_id(annotation.id, "test_tag", schedule_in=MINUS_5_MIN_IN_SECS)
-        # Change the job's expires_at date so that it's expired.
-        job = db_session.query(Job).one()
-        job.expires_at = now - datetime_.timedelta(minutes=1)
+        job = factories.SyncAnnotationJob(expires_at=now - datetime_.timedelta(hours=1))
 
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {}
         batch_indexer.index.assert_not_called()
-        assert db_session.query(Job).all() == [job]
+        assert job in db_session.query(Job).all()
 
-    @pytest.mark.usefixtures("with_indexed_annotations")
     def test_if_the_job_has_force_True_it_indexes_the_annotation_and_deletes_the_job(
-        self, annotation_ids, add_all, batch_indexer, db_session, queue
+        self, batch_indexer, db_session, factories, queue
     ):
-        add_all(force=True)
+        job = factories.SyncAnnotationJob(force=True)
 
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {
-            Queue.Result.SYNCED_FORCED: LIMIT,
-            Queue.Result.COMPLETED_FORCED: LIMIT,
-            Queue.Result.SYNCED_TOTAL: LIMIT,
-            Queue.Result.COMPLETED_TOTAL: LIMIT,
+            Queue.Result.SYNCED_FORCED: 1,
+            Queue.Result.COMPLETED_FORCED: 1,
+            Queue.Result.SYNCED_TOTAL: 1,
+            Queue.Result.COMPLETED_TOTAL: 1,
         }
-        assert db_session.query(Job).all() == []
-        batch_indexer.index.assert_called_once_with(
-            Any.list.containing(annotation_ids).only()
-        )
+        assert job not in db_session.query(Job)
```

Comment:
This line was asserting that the queue was empty, but what it actually meant to assert was just that this one particular `job` is no longer in the queue because its been deleted. Adding noise jobs to the queue (see below) forced me to fix the assertion. Lots of assertions-that-the-queue-is-empty below also had to be fixed in the same way. There are various ways to assert that a given `job` isn't in the DB with SQLAlchemy, I just went with this one